src/test/java/org/broadinstitute/hellbender/tools/PrintReadsIntegrationTest.java

```diff
@@ -39,4 +52,66 @@ public void testNoConflictPG() throws IOException {
+        final List<SimpleInterval> largeFileIntervals = Arrays.asList(new SimpleInterval("3",1_000_000, 1_000_001),
+               new SimpleInterval("3", 1_000_003, 1_000_100),
+                new SimpleInterval("20", 1_099_000, 1_100_000));
+
+        final List<SimpleInterval> cramIntervals = Arrays.asList(new SimpleInterval("20", 1_099_000, 1_100_000));
+        return new Object[][]{
+                {bucketPathToPublicHttpUrl(bam), bucketPathToPublicHttpUrl(bai), bam, bai, largeFileIntervals},
+                {toSignedUrl(bam), toSignedUrl(bai), bam, bai, largeFileIntervals},
+                {bucketPathToPublicHttpUrl(cram), bucketPathToPublicHttpUrl(crai), cram, crai, cramIntervals},
+                {toSignedUrl(cram), toSignedUrl(crai), cram, crai, cramIntervals}
+
+        };
+    }
+
+    @Test(groups = {"cloud", "bucket"}, dataProvider = "getHttpPaths")
+    public void testHttpPaths(String reads, String index, String nonHttpReads, String nonHttpIndex, List<SimpleInterval> intervals) throws IOException {
+        final ArgumentsBuilder args = new ArgumentsBuilder();
+        final File out = createTempFile("out", ".bam");
+        System.out.println("reading http");
+        args.addInput(reads)
+                .add(StandardArgumentDefinitions.CLOUD_PREFETCH_BUFFER_LONG_NAME, 1)
+                .add(StandardArgumentDefinitions.CLOUD_INDEX_PREFETCH_BUFFER_LONG_NAME, 1)
+                .add("read-index", index)
+                .addReference(GATKBaseTest.b37Reference)
+                .addOutput(out);
+        intervals.forEach(args::addInterval);
+        runCommandLine(args);
+
+        System.out.println("reading gs");
+        final ArgumentsBuilder args2 = new ArgumentsBuilder();
+        final File out2 = createTempFile("out", ".bam");
+        args2.addInput(nonHttpReads)
+                .add("read-index", nonHttpIndex)
+                .add(StandardArgumentDefinitions.CLOUD_PREFETCH_BUFFER_LONG_NAME, 1)
+                .add(StandardArgumentDefinitions.CLOUD_INDEX_PREFETCH_BUFFER_LONG_NAME, 1)
+                .addReference(GATKBaseTest.b37Reference)
+                .addOutput(out2);
+        intervals.forEach(args2::addInterval);
+        runCommandLine(args2);
+
+        SamAssertionUtils.assertEqualBamFiles(out, out2, false, ValidationStringency.DEFAULT_STRINGENCY);
+    }
+
+    public static String toSignedUrl(String path) {
+        final Storage storage = StorageOptions.getDefaultInstance().getService();
+        final BlobInfo info = BlobInfo.newBuilder(BucketUtils.getBucket(path), BucketUtils.getPathWithoutBucket(path)).build();
+        return storage.signUrl(info, 1L, TimeUnit.HOURS,
+                Storage.SignUrlOption.httpMethod(HttpMethod.GET)
+        ).toString();
+    }
+
+    public static String bucketPathToPublicHttpUrl(String path){
+        return String.format("https://storage.googleapis.com/%s/%s", BucketUtils.getBucket(path), BucketUtils.getPathWithoutBucket(path));
```

Comment:
If this method is generally useful, perhaps it should live in `BucketUtils` rather than in this test class.

Reply:
Sure, although it's not necessarily stable forever.  I don't know if they promise anywhere to construct paths like this, this is just what seems to work.